### DISCUSSION TITLE
MemoryStoreConfig Issue

### DISCUSSION THREAD

**Discussion by ronak-supraoracles** [libp2p/rust-libp2p#2305] [Q&A] [answered]
I am facing one issue. Please help me for that

In Libp2p I am putting data to DHT. When I am sharing any transaction object like this

```
{
        "transaction_id": "OkybUNAVLzYh3QouA24oLS3GhtRSBl",
        "from_account": "aca46912e9ad4e7b078a9f5d978ed599f599d0e47dccc403d06564da6e07720a",
        "to_account": "9b870ad21081be1914136549a97a8beb7ca9c6a97bee06dcd7f8e76fbd495e7e",
        "amount": 511,
        "signature": "ac1398e15263a9d7a48c118404c928c66573ca82ffcf0e101ad2d65b868054f01c5704bf4e1a70ba346d68de9d71b07944456b74967ae54b07b4caf47dfd210e"
      }
```

I can store one transaction under one key. But I want to store the multiple txn data under one key.
I can store upto 40 transaction under same key. But I want to more than that

**Comment by thomaseizinger** [comment, marked as answer]
This may be less of a problem with the DHT and the rust-libp2p specific implementation and more with how you are designing your data structures.

A DHT is by definition a key-value store. There can only be one value per key. You need to design your keys in a way such that each has only one value.

**Reply by ronak-supraoracles** [reply]
Thanks for reply @thomaseizinger 

Actually I am putting vector of array as one value. So vector count greater than 40 is not allowed. Can you please explain how we can set data structure so I can achieve that

**Reply by ronak-supraoracles** [reply]
I am trying to store vector of array as value inside but found issue that I can store only 40 transaction inside single array. If I tried to store more than that It returning the Quorum Failed Error. Can you please help me to resolve this.

**Reply by thomaseizinger** [reply]
Okay, storing a list should be fine!

Might be an issue with the maximum size of the value? @mxinden should know more here.

Have you tried increasing the max packet size?

https://docs.rs/libp2p-kad/0.31.0/libp2p_kad/struct.KademliaConfig.html#method.set_max_packet_size

**Reply by ronak-supraoracles** [reply]
Thanks for the wonderful help. Solution work for my issue. Thanks a lot
